src/nfts/NftImage.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+import { NftEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import SkeletonPlaceholder from 'src/components/SkeletonPlaceholder'
+import { Nft, NftOrigin } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import variables from 'src/styles/variables'
+import Logger from 'src/utils/Logger'
+
+const DEFAULT_IMAGE_HEIGHT = 360
+
+interface ImagePlaceHolderProps {
+  testID: string
+  height?: number
+  width?: number
+  borderRadius?: number
+}
+
+interface Props extends ImagePlaceHolderProps {
+  nft: Nft
+  origin: NftOrigin
+  shouldAutoScaleHeight?: boolean
+  ErrorComponent: React.ReactNode
+}
+
+function ImagePlaceholder({ height = 40, width, borderRadius = 0, testID }: ImagePlaceHolderProps) {
+  return (
+    <SkeletonPlaceholder
+      borderRadius={borderRadius}
+      backgroundColor={colors.gray2}
+      highlightColor={colors.white}
+    >
+      <View
+        style={{
+          height,
+          width: width ?? variables.width,
+          zIndex: -1,
+        }}
+        testID={testID ?? 'NftsInfoCarousel/ImagePlaceholder'}
+      />
+    </SkeletonPlaceholder>
+  )
+}
+
+export default function NftImage({
+  nft,
+  height,
+  width,
+  shouldAutoScaleHeight,
+  borderRadius,
+  origin,
+  ErrorComponent,
+  testID,
+}: Props) {
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState(!nft.metadata)
+  const [scaledHeight, setScaledHeight] = useState(DEFAULT_IMAGE_HEIGHT)
+
+  const imageUrl = useMemo(
+    () => nft.media.find((media) => media.raw === nft.metadata?.image)?.gateway,
+    [nft]
+  )
+
+  useEffect(() => {
+    setError(!nft.metadata)
+  }, [nft])
+
+  function sendEventWithError(error: boolean) {
+    const { contractAddress, tokenId } = nft
+    ValoraAnalytics.track(NftEvents.nft_image_load, {
+      tokenId,
+      contractAddress,
+      url: imageUrl,
+      origin,
+      error,
+    })
+
+    if (error) {
+      Logger.error(
+        origin,
```

Comment:
Don't we want the error?
```suggestion
        error,
```

Reply:
ah now I see the error is not an error but a boolean 🤔

Reply:
i made this a string. i think it's helpful to differentiate between when the load fails because of lack of metadata vs bad image url. and would be good to measure the ratio of successful vs failed image loads so i kept the event unchanged (was tempted to only capture the failures for a while)